Lib/unittest/mock.py

```diff
@@ -1242,14 +1242,25 @@ def _importer(target):
     return thing
 
 
+# _check_spec_arg_typos takes kwargs from commands like patch and checks that
+# they don't contain common misspellings of arguments related to autospeccing.
+def _check_spec_arg_typos(kwargs_to_check):
+    typos = ["autospect", "auto_spec", "set_spec"]
+    for typo in typos:
+        if typo in kwargs_to_check:
+            raise RuntimeError(
+                f"{typo} might be a typo; use unsafe=True if this is intended"
+            )
+
+
 class _patch(object):
 
     attribute_name = None
     _active_patches = []
 
     def __init__(
             self, getter, attribute, new, spec, create,
-            spec_set, autospec, new_callable, kwargs
+            spec_set, autospec, new_callable, unsafe, kwargs
```

Comment:
code style wise, this is already a very long list of arguments.  when adding another one, I'd add it as a keyword only argument at the end.

`, kwargs, *, unsafe):`

Reply:
Thanks for teaching me about keyword-only arguments. This also allowed me to drop some unnecessary changes done previously just to supply the positional argument correctly.